### PR TITLE
Fixing `NotSerializableException: org.acegisecurity.context.SecurityContext$1`

### PR DESCRIPTION
**Pull request by jglick**
While testing some CloudBees CI code in a non-production context involving serialized web sessions, I noticed an error

```
java.io.NotSerializableException: org.acegisecurity.context.SecurityContext$1
```

Looking into #4848 I noticed that I had forgotten `implements Serializable` in a spot:

```console
$ javap -classpath ~/.m2/repository/org/acegisecurity/acegi-security/1.0.7/acegi-security-1.0.7.jar org.acegisecurity.context.SecurityContext
Compiled from "SecurityContext.java"
public interface org.acegisecurity.context.SecurityContext extends java.io.Serializable {
  public abstract org.acegisecurity.Authentication getAuthentication();
  public abstract void setAuthentication(org.acegisecurity.Authentication);
}
```

### Testing done

Test without patch:

```
java.io.NotSerializableException: org.acegisecurity.context.SecurityContext$1
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1175)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:345)
	at org.acegisecurity.context.SecurityContextTest.serDeser(SecurityContextTest.java:61)
	at org.acegisecurity.context.SecurityContextTest.serializabilityFromSpring(SecurityContextTest.java:44)
java.io.NotSerializableException: org.acegisecurity.context.SecurityContextImpl
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1175)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1543)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1500)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1423)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1169)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:345)
	at org.acegisecurity.context.SecurityContextTest.serDeser(SecurityContextTest.java:61)
	at org.acegisecurity.context.SecurityContextTest.serializabilityToSpring(SecurityContextTest.java:54)
```

### Proposed changelog entries

- N/A, not known to affect production code

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/org/acegisecurity/context/SecurityContext.java
Comment (jglick) [inline core/src/main/java/org/acegisecurity/context/SecurityContext.java:58]: ignore ws

**Review by timja — APPROVED** [review]

**Review by julieheard — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/java/org/acegisecurity/context/SecurityContext.java
Comment (jglick) [inline core/src/main/java/org/acegisecurity/context/SecurityContext.java:39]: Note that just this line suffices to make the tests pass. I refactored the inner classes to member classes since it is not a good practice to use inner classes for anything that might be serialized.

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**
